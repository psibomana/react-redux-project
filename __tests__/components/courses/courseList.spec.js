import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import CourseList from '../../../src/components/course/courseList'

describe('Course List Test', () => {
  let wrapper, courses, addCourse, deleteCourse, updateCourse, listCourses;

  beforeEach(() => {

    addCourse = jest.fn();
    deleteCourse = jest.fn();
    updateCourse = jest.fn();
    listCourses = jest.fn();

    courses = [
      {
        id: "react-flux-building-applications",
        title: "Building Applications in React and Flux",
        watchHref: "http://www.pluralsight.com/courses/react-flux-building-applications",
        authorId: "cory-house",
        length: "5:08",
        category: "JavaScript"
      },
      {
        id: "clean-code",
        title: "Clean Code: Writing Code for Humans",
        watchHref: "http://www.pluralsight.com/courses/writing-clean-code-humans",
        authorId: "cory-house",
        length: "3:10",
        category: "Software Practices"
      }
    ]

    wrapper = shallow(
      <CourseList
        courses={courses}
        addCourse={addCourse}
        deleteCourse={deleteCourse}
        updateCourse={updateCourse}
        listCourses={listCourses}/>
    )
  })

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
