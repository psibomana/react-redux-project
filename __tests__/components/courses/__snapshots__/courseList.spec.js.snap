// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course List Test should render as expected 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Â 
      </th>
      <th>
        Title
      </th>
      <th>
        Author
      </th>
      <th>
        Category
      </th>
      <th>
        Length
      </th>
    </tr>
  </thead>
  <tbody>
    <CourseListRow
      course={
        Object {
          "authorId": "cory-house",
          "category": "JavaScript",
          "id": "react-flux-building-applications",
          "length": "5:08",
          "title": "Building Applications in React and Flux",
          "watchHref": "http://www.pluralsight.com/courses/react-flux-building-applications",
        }
      }
      key="react-flux-building-applications"
    />
    <CourseListRow
      course={
        Object {
          "authorId": "cory-house",
          "category": "Software Practices",
          "id": "clean-code",
          "length": "3:10",
          "title": "Clean Code: Writing Code for Humans",
          "watchHref": "http://www.pluralsight.com/courses/writing-clean-code-humans",
        }
      }
      key="clean-code"
    />
  </tbody>
</table>
`;
